scipy/cluster/tests/test_vq.py

```diff
@@ -386,16 +379,16 @@ def test_kmeans2_kpp_high_dim(self, xp):
 
         data = xp.asarray(data)
         res, _ = kmeans2(data, xp.asarray(2), minit='++')
-        assert_array_almost_equal(res, centers, decimal=0)
+        xp_assert_close(res, xp.asarray(centers), rtol=0.2)
 
     @skip_if_array_api_gpu
     @array_api_compatible
     def test_kmeans_diff_convergence(self, xp):
         # Regression test for gh-8727
         obs = xp.asarray([-3, -1, 0, 1, 1, 8], dtype=xp.float64)
         res = kmeans(obs, xp.asarray([-3., 0.99]))
-        assert_allclose(res[0], xp.asarray([-0.4,  8.]))
-        assert_allclose(res[1], 1.0666666666666667)
+        xp_assert_close(res[0], xp.asarray([-0.4,  8.]), check_dtype=False)
```

Comment:
`res[0]` is `float64`, but `xp.asarray([-0.4,  8.])` is `float32` for `torch`.

Reply:
The test requiring `res` to have `float64` dtype when the input to `kmeans` is also `float64` seems like a perfectly reasonable requirement and good to test. I'd probably consider anything else a bug.

Reply:
No worries. It's also not just PyTorch - all deep learning focused libraries will default to `float32`, because that's much better supported on GPU/TPU, and more than enough precision for deep learning.

Reply:
Right. I knew it was useful, I just didn't know it was the default. 

Here was the overall comment btw. https://github.com/scipy/scipy/pull/19251#pullrequestreview-1630077699